### PR TITLE
feat: referendum voting status

### PR DESCRIPTION
**Pull request by johnthecat**

### CODE REVIEW

File: src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
 import { detailsAggregate } from '../../aggregates/details';
 import { ProposalDescription } from './ProposalDescription';
 import { VotingStatus } from './VotingStatus';
 import { DetailsCard } from './DetailsCard';
+import { AggregatedReferendum } from '../../types/structs';
 
 type Props = {
   chain: Chain;
-  referendum: Referendum;
+  referendum: AggregatedReferendum;
```
Comment (Asmadek) [inline src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx:16]: Maybe call it in another way and extract `referendum` to exclude using of `referendum.referendum`?

File: src/renderer/features/governance/components/ReferendumDetails/VoteDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    </button>
+  );
+};
+
+const AboutVoting = () => {
+  const { t } = useI18n();
+
+  return (
+    <div className="flex flex-col gap-2 p-4 whitespace-pre-line">
+      <Trans
```
Comment (Asmadek) [inline src/renderer/features/governance/components/ReferendumDetails/VoteDialog.tsx]: Maybe better to split it for small chunks? 

```ts
<SmallTittleText>{t('governance.voting.aboutTitle')}</SmallTittleText>
<FootnoteText>
    <Trans t={t}
        i18nKey="governance.voting.about"
        components={{
          ul: <ul className="list-disc ml-4" />,
          li: <li />,
        }} />
</FootnoteText>
```

File: src/renderer/features/governance/components/ReferendumDetails/VoteDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        t={t}
+        i18nKey="governance.voting.about"
+        components={{
+          header: <SmallTitleText />,
+          p: <div className="text-footnote text-text-secondary" />,
```
Comment (Asmadek) [inline src/renderer/features/governance/components/ReferendumDetails/VoteDialog.tsx]: we have `<FootnoteText>` for such text

File: src/renderer/features/governance/components/ReferendumDetails/VotingSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <FootnoteText>{t('governance.referendum.aye')}</FootnoteText>
+        <FootnoteText className="text-end grow">
+          {t('governance.referendum.votes', { votes: ayeBalance.suffix })}
+        </FootnoteText>
+      </div>
+      <div className="flex gap-2 items-center w-full">
+        <div className="w-1 h-3 rounded-[4px] bg-icon-negative" />
+        <FootnoteText>{t('governance.referendum.nay')}</FootnoteText>
+        <FootnoteText className="text-end grow">
+          {/*{t('governance.referendum.votes', { votes: naysBalance.value + naysBalance.suffix })}*/}
```
Comment (Asmadek) [inline src/renderer/features/governance/components/ReferendumDetails/VotingSummary.tsx]: do you need this commented code?

File: src/renderer/shared/api/governance/off-chain/service/polkassemblyService.ts [hunk 5]
```diff
@@ -1,84 +1,77 @@
...
+      },
+    )
+    .then(mapListingPost);
+};
 
-    const ping: PolkassemblyData = await fetch(getApiUrl(1), { method: 'GET', headers }).then((r) => r.json());
-    const totalPages = Math.ceil(ping.count / pageSize);
+const getReferendumVotes: GovernanceApi['getReferendumVotes'] = (chain, referendumId, callback) => {
+  const mapDecision = (x: PostVote['decision']): ReferendumVote['decision'] => {
+    switch (x) {
```
Comment (Asmadek) [inline src/renderer/shared/api/governance/off-chain/service/polkassemblyService.ts]: maybe use object for this?

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx (hunk 1 above)
Comment (johnthecat) [reply src/renderer/features/governance/components/ReferendumDetails/ReferendumDetailsDialog.tsx:16]: Merged raferendum with additional fields directly
```ts
referendum & {
 /* ... */
}
```

**Pull request merged**